### ISSUE TITLE
$select.selected gets wrong values for uiSelectMultiple

### ISSUE THREAD

**Issue by FarSeeing** [angular-ui/ui-select#1049]
[uiSelectMultipleDirective.js#L143](https://github.com/angular-ui/ui-select/blob/2d1422d109dc1cd64213271e59506b14f74d356f/src/uiSelectMultipleDirective.js#L143)

```javascript
        if(!angular.isArray(ngModel.$viewValue)){
          // Have tolerance for null or undefined values
          if(angular.isUndefined(ngModel.$viewValue) || ngModel.$viewValue === null){
            $select.selected = [];
          } else {
            throw uiSelectMinErr('multiarr', "Expected model value to be array but got '{0}'", ngModel.$viewValue);
          }
        }
        $select.selected = ngModel.$viewValue;
```

The check for `ngModel.$viewValue` being `null` or `undefined` makes no sense now as `$select.selected` will still get the original value of `ngModel.$viewValue`.

**Comment by awerlang** [comment]
+1

**Comment by nesinervink** [comment]
+1

This code needs testing, but seems like it solves all the issues with `$select.selected` being `undefined`, because on `$render` `$select.selected` is assigned model's `$viewValue` before `$watch` in [uiSelectSingleDirective.js#L42](https://github.com/angular-ui/ui-select/blob/e75bf4906da0047ba837b6db539b0feecc3b327b/src/uiSelectSingleDirective.js#L42) sets the `$viewValue` to empty array in case `$select.selected` was `undefined`.

nesinervink/ui-select@b237a42ad2de95522448555080645b31dcc50aef

```javascript
ngModel.$render = function() {
  // Make sure that model value is array
  if(!angular.isArray(ngModel.$viewValue)){
    // Have tolerance for null or undefined values
    if(angular.isUndefined(ngModel.$viewValue) || ngModel.$viewValue === null){
      $select.selected = [];
    } else {
      throw uiSelectMinErr('multiarr', "Expected model value to be array but got '{0}'", ngModel.$viewValue);
    }
  } else {
    $select.selected = ngModel.$setViewValue;
  }
  
  $selectMultiple.refreshComponent();
  scope.$evalAsync(); //To force $digest
};
```

**Comment by nesinervink** [comment]
Problem persists in v0.18.0 and this push still solves it.

**Comment by user378230** [comment]
This should be resolved by #1709, please update this issue if it does not (when the next version is released).
